modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/osgi/commands/PortalAddressOSGiCommands.java

```diff
@@ -128,6 +128,86 @@ public void populateCompanyCountries(long companyId) throws Exception {
 		}
 	}
 
+	public void repopulateCompanyCountries(long companyId) throws Exception {
+		Company company = _companyLocalService.getCompany(companyId);
+
+		if (_log.isDebugEnabled()) {
+			_log.debug("Reinitializing countries for company " + companyId);
+		}
+
+		JSONArray countriesJSONArray = _getJSONArray(
+			"com/liferay/address/dependencies/countries.json");
+
+		List<Country> countries = _countryLocalService.getCompanyCountries(
+			companyId);
+
+		HashSet<String> countryNames = new HashSet<>();
+
+		for (Country country : countries) {
+			countryNames.add(country.getName());
+		}
+
+		for (int i = 0; i < countriesJSONArray.length(); i++) {
+			JSONObject countryJSONObject = countriesJSONArray.getJSONObject(i);
+
+			try {
+				String name = countryJSONObject.getString("name");
+
+				if (countryNames.contains(name)) {
+					Country country = _countryLocalService.getCountryByName(
+						companyId, name);
+
+					country = _countryLocalService.updateCountry(
+						country.getCountryId(),
+						countryJSONObject.getString("a2"),
+						countryJSONObject.getString("a3"), country.isActive(),
+						country.isBillingAllowed(),
+						countryJSONObject.getString("idd"), name,
+						countryJSONObject.getString("number"),
+						country.getPosition(), country.isShippingAllowed(),
+						country.isSubjectToVAT());
+
+					_processCountryRegions(country);
+				}
+				else {
+					ServiceContext serviceContext = new ServiceContext();
```

Comment:
Have you already considered refactoring this part of the new OSGi command?
Since the `else` clause of `repopulateCompanyCountries(...)` is based on a part of `populateCompanyCountries(...)`, it may be a good idea to create a private method so we can avoid duplicate code.

Reply:
You're right! Thanks for the suggestion, I've done as you've requested and restarted the test suite as well.